hudi-client/hudi-java-client/src/main/java/org/apache/hudi/execution/bulkinsert/JavaCustomColumnsSortPartitioner.java

```diff
@@ -40,8 +42,11 @@
   private final String[] sortColumnNames;
   private final Schema schema;
   private final boolean consistentLogicalTimestampEnabled;
+  private final boolean isSorted;
 
-  public JavaCustomColumnsSortPartitioner(String[] columnNames, Schema schema, boolean consistentLogicalTimestampEnabled) {
+  public JavaCustomColumnsSortPartitioner(String[] columnNames, Schema schema, boolean consistentLogicalTimestampEnabled, HoodieWriteConfig config) {
+    String partitionPath = config.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME);
```

Comment:
This could actually be multiple columns. Let's instead create a KeyGenerator and fetch `getPartitionPathFields` from it.  You can create it using `HoodieSparkKeyGeneratorFactory.createKeyGenerator`

Reply:
That's in hudi-spark-client so IDK if we can use that in hudi-java-client. We don't currently get that dependency

Reply:
Ha! My mind completely ignored Spark in the name. Conceptually though this factory should be generic and there's no reason for it to be bound to Spark.

Reply:
For now then, let's extract the utility parsing partition-path fields from `ComplexKeyGenerator`